lib/web_ui/test/golden_tests/engine/canvas_draw_image_golden_test.dart

```diff
@@ -123,13 +129,140 @@ void main() async {
     double testHeight = testImage.height.toDouble();
     rc.drawImageRect(testImage, Rect.fromLTRB(testWidth / 2, 0, testWidth, testHeight),
         Rect.fromLTRB(100, 30, 2 * testWidth, 2 * testHeight), new Paint());
+    rc.restore();
     await _checkScreenshot(rc, 'draw_image_rect_with_transform_source');
   });
+
+  // Regression test for https://github.com/flutter/flutter/issues/44845
+  // Circle should draw on top of image not below.
+  test('Paints on top of image', () async {
```

Comment:
We should also have a test that checks that we can still paint _under_ the image :)

Reply:
Done. https://github.com/flutter/goldens/pull/69